ungleich/dynamicweb
# Task/3622/decouple opennebula from DCL's purchase VM flow

This PR tries to move the create vm xml-rpc call made in the DCL VM purchase flow into a celery asynchronous task, thereby decoupling the dependence of the flow on OpenNebula.

Prerequisites:
1. Install the newly added requirements `pip -r requirements.txt`
2. Configure .env to include the following parameters
```
CELERY_BROKER_URL='redis+socket:///path/to/redis/unix/socket'
CELERY_RESULT_BACKEND='redis+socket:///path/to/redis/unix/socket'
CELERY_MAX_RETRIES=5
```

To Test:
1. Change the username/password used to connect to OpenNebula. Put some false values so that we are sure, the OpenNebula is not accessible.
2. Make sure that OpenNebula is inaccessible by logging into hosting app. Non of the OpenNebula features (creating, listing or deleting) VMs should be possible.
3. Try buying a VM from Data Center Light's landing page calculator. The flow should work fine.
